precompiles/src/ed25519.rs

```diff
@@ -31,11 +31,13 @@ pub fn verify(
         let start = i
             .saturating_mul(SIGNATURE_OFFSETS_SERIALIZED_SIZE)
             .saturating_add(SIGNATURE_OFFSETS_START);
-        let end = start.saturating_add(SIGNATURE_OFFSETS_SERIALIZED_SIZE);
 
-        // bytemuck wants structures aligned
-        let offsets: &Ed25519SignatureOffsets = bytemuck::try_from_bytes(&data[start..end])
-            .map_err(|_| PrecompileError::InvalidDataOffsets)?;
+        // SAFETY:
+        // - data[start..] is guaranteed to be >= size of Ed25519SignatureOffsets
+        // - Ed25519SignatureOffsets is a POD type, so we can safely read it as an unaligned struct
+        let offsets = unsafe {
```

Comment:
might be nice to pull this into a function that returns a generic so we could light it up with comments and debug asserts and such without cluttering the precompile files. Not going to gate this PR on it tho